back/sssscs/src/main/java/com/ib/pki/manual/KeyUtil.java

```diff
@@ -0,0 +1,155 @@
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.bouncycastle.openssl.jcajce.JcaMiscPEMGenerator;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.util.io.pem.PemObjectGenerator;
+
+public class KeyUtil {
+	private static String DIR_KEYS = "./keys";
+	private static String DIR_CERTS = "./certs";
+	private static String SECRET = "HFJKEHF2-r-2dhUOI#_R0238r0-fwk3w=e2u8v f3h";
+	
+	////////////////////////////////////////////////////////////
+	// Filename generators
+	////////////////////////////////////////////////////////////
+
+	private String getFnameCert(String alias) {
+		return DIR_CERTS + "/" + alias + ".crt";
+	}
+	
+	private String getFnamePrivKey(String alias) {
+		return DIR_KEYS + "/" + alias + ".key";
+	}
+	
+	////////////////////////////////////////////////////////////
+	// Read and write keys/certs.
+	////////////////////////////////////////////////////////////
+	
+	public PrivateKey getPrivateKey(String alias) {
+		String fnamePriv = getFnamePrivKey(alias);
+		try {
+			if (!new File(fnamePriv).isFile()) {
+				return null;
+			}
+			FileInputStream is = new FileInputStream(fnamePriv);
+			byte[] key = Base64.getDecoder().decode(is.readAllBytes());
+	        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+	        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(key);
+	        PrivateKey finalKey = keyFactory.generatePrivate(keySpec);
+			
+	        return finalKey;
+		} 
+		catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	public PublicKey getPublicKey(String alias) {
+		X509Certificate cert = getX509Certificate(alias);
+		return cert.getPublicKey();
+	}
+	
+	public X509Certificate getX509Certificate(String alias) {
+		String fnameCert = getFnameCert(alias);
+		
+		try {
+			CertificateFactory fac = CertificateFactory.getInstance("X509");
+			FileInputStream is = new FileInputStream(fnameCert);
+			X509Certificate cert = (X509Certificate) fac.generateCertificate(is);
+			return cert;
+		} catch (CertificateException | FileNotFoundException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	public void savePrivateKey(String alias, PrivateKey privateKey) {
+		String fnamePriv = getFnamePrivKey(alias);
+		
+		try {	
+			byte[] key2 = privateKey.getEncoded();
+			FileOutputStream keyfos2 = new FileOutputStream(fnamePriv);
+			keyfos2.write(Base64.getEncoder().encodeToString(key2).getBytes());
+			keyfos2.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public void saveX509Certificate(String alias, X509Certificate certificate) {
+		String fnameCert = getFnameCert(alias);
+		
+		try {
+			PrintWriter out = new PrintWriter(fnameCert);
+			JcaPEMWriter pemWriter = new JcaPEMWriter(out);
+			PemObjectGenerator objGen = new JcaMiscPEMGenerator(certificate);
+			pemWriter.writeObject(objGen);
+			pemWriter.flush();
+			pemWriter.close();
```

Comment:
Use a try-with maybe?

Reply:
Fixed in [b111018](https://github.com/magley/sssscs/pull/5/commits/b111018d31a0b6a52e6ee80782ef789c34c47a82).